File: web_app/src/app/entities/frame-data.ts
/***
** Copyright (C) 2020-2024 neuRay Labs. All rights reserved.
**
** The information and source code contained herein is the exclusive 
** property of neuRay Labs and may not be disclosed, examined, reproduced, redistributed, used in source and binary forms, in whole or in part  
** without explicit written authorization from the company.
**
***/

export interface FrameData {
    device_id: string
    device_name: string
    timestamp: string

    tracks: TrackData[]
    points: PointData[]
    targets_index: number[]

    targets_height: TargetsHeight[]

    vital_signs? : VitalSignsInfo
}

export interface TrackData {
    track_id: number
    position_x : number
    position_y : number
    position_z : number
    velocity_x : number
    velocity_y : number
    velocity_z : number
    acceleration_x : number
    acceleration_y : number
    acceleration_z : number
}

export interface PointData {
    range: number
    azimuth : number
    elevation : number
    doppler : number
    snr : number
    position_x : number
    position_y : number
    position_z : number
}

export interface TargetsHeight {
    target_id : number
    max_z : number
    min_z : number
}

export interface VitalSignsInfo {
    target_id : number
    range_bin : number
    breathing_deviation : number
    heart_rate : number
    breathing_rate : number
    heart_circular_buffer : number[]
    breath_circular_buffer : number[]
}

File: web_service/src/Tracking/FrameData.cs
/***
** Copyright (C) 2020-2023 neuRay Labs. All rights reserved.
**
** The information and source code contained herein is the exclusive 
** property of neuRay Labs and may not be disclosed, examined, reproduced, redistributed, used in source and binary forms, in whole or in part  
** without explicit written authorization from the company.
**
***/
using System.Text.Json.Serialization;

namespace WebService.Tracking;

public class FrameData 
{
    public class Point
    {
        [JsonPropertyName("range")]
        public float Range { get; set; }

        [JsonPropertyName("azimuth")]
        public float Azimuth { get; set; }

        [JsonPropertyName("elevation")]
        public float Elevation { get; set; }

        [JsonPropertyName("doppler")]
        public float Doppler { get; set; }

        [JsonPropertyName("snr")]
        public float SNR { get; set; }

        [JsonPropertyName("position_x")]
        public float PositionX { get; set; }       

        [JsonPropertyName("position_y")]
        public float PositionY { get; set; }       

        [JsonPropertyName("position_z")]
        public float PositionZ { get; set; }       
    }

    public class Track
    {
        [JsonPropertyName("track_id")]
        public uint TrackId { get; set; }

        [JsonPropertyName("position_x")]
        public float PositionX { get; set; }

        [JsonPropertyName("position_y")]
        public float PositionY { get; set; }

        [JsonPropertyName("position_z")]
        public float PositionZ { get; set; }

        [JsonPropertyName("velocity_x")]
        public float VelocityX { get; set; }

        [JsonPropertyName("velocity_y")]
        public float VelocityY { get; set; }

        [JsonPropertyName("velocity_z")]
        public float VelocityZ { get; set; }

        [JsonPropertyName("acceleration_x")]
        public float AccelerationX { get; set; }

        [JsonPropertyName("acceleration_y")]
        public float AccelerationY { get; set; }

        [JsonPropertyName("acceleration_z")]
        public float AccelerationZ { get; set; }
    }

    [JsonPropertyName("points")]
    public List<Point> PointsList { get; set; } = new List<Point>();

    [JsonPropertyName("targets_index")]
    public List<byte> TargetsIndexList { get; set; } = new List<byte>();

    [JsonPropertyName("device_id")]
    public string DeviceId { get; set; } = String.Empty;
    
    [JsonPropertyName("device_name")]
    public string DeviceName { get; set; } = String.Empty;
    
    [JsonPropertyName("timestamp")]
    public DateTime Timestamp { get; set; } = DateTime.UtcNow;

    [JsonPropertyName("tracks")]
    public List<Track> tracksList { get; set; } = new List<Track>();
}